File: frontend/src/features/bean-defect-detection/config/sensorQualityConfig.js
// =========================================================
// SENSOR QUALITY CONFIGURATION
// =========================================================
//
// Response / Delta:
//     Sample - Baseline
//
// IMPORTANT:
// Keep these values synchronized with backend:
// sensor_quality_config.py
//
// =========================================================

// =========================================================
// SENSOR THRESHOLDS
// =========================================================

export const SENSOR_THRESHOLDS = {
  mq2: {
    badThreshold: 70,
    direction: "high",
  },

  mq3: {
    badThreshold: 28,
    direction: "high",
  },

  mq135: {
    badThreshold: 9.5,
    direction: "high",
  },

  moisture: {
    badThreshold: -16.0,
    direction: "low",
  },

  humidity: {
    badThreshold: 10.7,
    direction: "high",
  },
};

// =========================================================
// SENSOR VOTING CONFIGURATION
// =========================================================

export const TOTAL_VOTING_SENSORS = 5;

export const SENSOR_VOTE_WEIGHT = 20;

// =========================================================
// SENSOR STATUS RULES
// =========================================================

export const GOOD_MAX_BAD_VOTES = 1;

export const REVIEW_BAD_VOTES = 2;

export const BAD_MIN_BAD_VOTES = 3;

// =========================================================
// TEMPERATURE
// =========================================================
//
// Temperature is supporting environmental information only.
// It does NOT participate in GOOD / REVIEW / BAD voting.
//
// =========================================================

export const TEMPERATURE_USED_FOR_DECISION = false;

File: backend/app/modules/bean_defect_detection/sensor_quality_config.py
# =========================================================
# SENSOR QUALITY CONFIGURATION
# =========================================================
#
# Response / Delta:
#     sample - baseline
#
# Research-defined thresholds derived from the
# collected GOOD / BAD coffee bean experiments.
#
# =========================================================


# =========================================================
# SENSOR THRESHOLDS
# =========================================================

MQ2_BAD_THRESHOLD = 70

MQ3_BAD_THRESHOLD = 28

MQ135_BAD_THRESHOLD = 9.5

MOISTURE_BAD_THRESHOLD = -16.0

HUMIDITY_BAD_THRESHOLD = 10.7


# =========================================================
# SENSOR VOTING CONFIGURATION
# =========================================================
#
# Five sensors participate:
#
# MQ-2
# MQ-3
# MQ-135
# Moisture
# Humidity
#
# Temperature is supporting information only.
#
# =========================================================

TOTAL_VOTING_SENSORS = 5

SENSOR_VOTE_WEIGHT = 20.0


# =========================================================
# SENSOR STATUS RULES
# =========================================================
#
# 0-1 BAD votes -> GOOD
# 2 BAD votes   -> REVIEW
# 3-5 BAD votes -> BAD
#
# =========================================================

GOOD_MAX_BAD_VOTES = 1

REVIEW_BAD_VOTES = 2

BAD_MIN_BAD_VOTES = 3